src/init/pkg_installer_mac.sh

```diff
@@ -121,10 +141,15 @@ else
     rm -rf ./backup/restore
     mkdir -p ./backup/restore
     tar xzf ./backup/backup_[${BDATE}].tar.gz -C ./backup/restore >> ./logs/upgrade.log 2>&1
-
-    for dir in ./backup/restore/*; do
-        cp -a ${dir}/* /$(basename ${dir}) >> ./logs/upgrade.log 2>&1
-    done
+    RESULT=$?
```

Comment:
Finally I have included the same comparison for this `tar` command. I left it out because I felt that it did not have the same impact in the case of macOS, since it is not necessary to copy any files afterwards, and there was no possibility of generating a chain error by copying system files. But I have decided to include it because at least we can avoid to continue with the execution if that decompression has gone wrong.